File: civitas/src/civitas/SorpresaPorCasaHotel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author Usuario
 */
public class SorpresaPorCasaHotel extends Sorpresa {
    
    private int valor;

    public SorpresaPorCasaHotel(int valor, String texto) {
        super(texto);
        this.valor = valor;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if(jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            todos.get(actual).modificarSaldo(valor * 
                    todos.get(actual).cantidadCasasHoteles());
        }
    }
    
    @Override
    public String toString() {
        String salida = "Valor: " + valor + super.toString();
        return salida;
    }

}

File: civitasRuby/pkg/civitas-0.0.1/lib/sorpresa_por_casa_hotel.rb
# encoding:utf-8
module Civitas

  class SorpresaPorCasaHotel < Sorpresa
    
    def initialize(texto, valor)
      super(texto)
      @valor = valor
    end
    
    def aplicar_a_jugador(actual, todos)
      if jugador_correcto(actual, todos)
        super(actual, todos)
        todos.at(actual).modificar_saldo(@valor * 
          todos.at(actual).cantidad_casas_hoteles)
      end
    end
  
    def to_string
      return super.to_string + "\nValor: #{@valor}"
    end
    
  public_class_method :new  
  end
end